File: tree-hugger/lib/tests/fixtures/signatures.scala
package example

/** A simple greeting function. */
def greet(name: String): String = {
  s"Hello, $name!"
}

/** A function with multiple parameters. */
def greetWithAge(name: String, age: Int): String = {
  s"Hello, $name ($age)!"
}

/** A function with default parameter. */
def greetWithPrefix(name: String, prefix: String = "Hello"): String = {
  s"$prefix, $name!"
}

/** A void function. */
def printGreeting(message: String): Unit = {
  println(message)
}

/** A function with varargs. */
def greetMany(names: String*): Unit = {
  names.foreach(name => println(s"Hello, $name"))
}

/** A class with methods. */
class Greeter(val prefix: String = "Hello") {
  /** A simple greeting method. */
  def greet(name: String): String = {
    s"$prefix, $name!"
  }
}

/** An object with methods. */
object DefaultGreeter {
  def greet(name: String): String = s"Hello, $name!"
}

File: tree-hugger/lib/tests/fixtures/signatures.cpp
#include <string>
#include <vector>

// A simple greeting function.
std::string greet(std::string name) {
    return "Hello, " + name + "!";
}

// A function with multiple parameters.
int add(int a, int b) {
    return a + b;
}

// A function with default parameter.
std::string greetWithPrefix(std::string name, std::string prefix = "Hello") {
    return prefix + ", " + name + "!";
}

// A class with methods.
class Greeter {
public:
    Greeter(std::string prefix);
    std::string greet(std::string name);
    void greetMany(std::vector<std::string> names);

private:
    std::string prefix_;
};

Greeter::Greeter(std::string prefix) : prefix_(prefix) {}

std::string Greeter::greet(std::string name) {
    return prefix_ + ", " + name + "!";
}

void Greeter::greetMany(std::vector<std::string> names) {
    for (const auto& name : names) {
        greet(name);
    }
}
